2024-07-11 12:36:03.665708: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:36:03.665888: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:36:12,205 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:36:12,317 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:36:12,318 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:36:12,318 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:36:13,405 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:36:13,405 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:13,421 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.4633333333333333}
INFO flwr 2024-07-11 12:36:13,421 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:36:13,754 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:36:13,776 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:36:13,778 | fedavg.py:242 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:13,785 | server.py:131 | fit progress: (1, 0.41330876975101233, {'accuracy': 0.8133333333333334}, 0.36349209999999843)
DEBUG flwr 2024-07-11 12:36:13,786 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:36:13,807 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:36:13,808 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:36:13,808 | server.py:227 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:36:13,834 | server.py:241 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:13,845 | server.py:131 | fit progress: (2, 0.41763526183377064, {'accuracy': 0.8033333333333333}, 0.4227287999999998)
DEBUG flwr 2024-07-11 12:36:13,845 | server.py:178 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:36:13,864 | server.py:192 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2024-07-11 12:36:13,865 | server.py:227 | fit_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:36:13,890 | server.py:241 | fit_round 3 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:13,901 | server.py:131 | fit progress: (3, 0.4176414399436547, {'accuracy': 0.8033333333333333}, 0.47939569999999954)
DEBUG flwr 2024-07-11 12:36:13,901 | server.py:178 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:36:13,924 | server.py:192 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:36:13,924 | server.py:227 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:36:13,953 | server.py:241 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:13,965 | server.py:131 | fit progress: (4, 0.4174367728148087, {'accuracy': 0.8066666666666666}, 0.5429967999999992)
DEBUG flwr 2024-07-11 12:36:13,965 | server.py:178 | evaluate_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:36:13,986 | server.py:192 | evaluate_round 4 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:36:13,986 | server.py:227 | fit_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:36:14,020 | server.py:241 | fit_round 5 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:14,031 | server.py:131 | fit progress: (5, 0.42199154333360184, {'accuracy': 0.7966666666666666}, 0.609518099999999)
DEBUG flwr 2024-07-11 12:36:14,032 | server.py:178 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:36:14,053 | server.py:192 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:36:14,053 | server.py:227 | fit_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:14,087 | server.py:241 | fit_round 6 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:14,100 | server.py:131 | fit progress: (6, 0.42249505553109745, {'accuracy': 0.79}, 0.6777824999999993)
DEBUG flwr 2024-07-11 12:36:14,100 | server.py:178 | evaluate_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:14,128 | server.py:192 | evaluate_round 6 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:36:14,129 | server.py:227 | fit_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:14,166 | server.py:241 | fit_round 7 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:14,179 | server.py:131 | fit progress: (7, 0.42251269240070827, {'accuracy': 0.79}, 0.7572786999999988)
DEBUG flwr 2024-07-11 12:36:14,179 | server.py:178 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:14,208 | server.py:192 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:36:14,209 | server.py:227 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:14,247 | server.py:241 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:14,260 | server.py:131 | fit progress: (8, 0.42251242719917526, {'accuracy': 0.79}, 0.8385663999999995)
DEBUG flwr 2024-07-11 12:36:14,261 | server.py:178 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:14,290 | server.py:192 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:36:14,291 | server.py:227 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:14,332 | server.py:241 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:14,345 | server.py:131 | fit progress: (9, 0.4225123444376344, {'accuracy': 0.79}, 0.9229021999999993)
DEBUG flwr 2024-07-11 12:36:14,345 | server.py:178 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:14,376 | server.py:192 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:36:14,376 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:14,417 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:14,430 | server.py:131 | fit progress: (10, 0.4225123322816466, {'accuracy': 0.79}, 1.0088263999999985)
DEBUG flwr 2024-07-11 12:36:14,431 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:14,461 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:36:14,461 | server.py:153 | FL finished in 1.0394577999999992
INFO flwr 2024-07-11 12:36:14,462 | app.py:225 | app_fit: losses_distributed [(1, 0.41729217519362766), (2, 0.4126543228824933), (3, 0.4202513058980306), (4, 0.42009749710559846), (5, 0.41142102407998055), (6, 0.41337017970425743), (7, 0.41337054584707533), (8, 0.4133705147675106), (9, 0.41337051349026815), (10, 0.4133705122130258)]
INFO flwr 2024-07-11 12:36:14,462 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:36:14,462 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:36:14,462 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.41330876975101233), (2, 0.41763526183377064), (3, 0.4176414399436547), (4, 0.4174367728148087), (5, 0.42199154333360184), (6, 0.42249505553109745), (7, 0.42251269240070827), (8, 0.42251242719917526), (9, 0.4225123444376344), (10, 0.4225123322816466)]
INFO flwr 2024-07-11 12:36:14,462 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.4633333333333333), (1, 0.8133333333333334), (2, 0.8033333333333333), (3, 0.8033333333333333), (4, 0.8066666666666666), (5, 0.7966666666666666), (6, 0.79), (7, 0.79), (8, 0.79), (9, 0.79), (10, 0.79)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.90486516,  0.85787907,  0.83130324,  0.17065442,  0.30312938,
         0.24800433, -0.04739454, -0.05923274]]), array([0.0272715])]
[array([[-1.90486516,  0.85787907,  0.83130324,  0.17065442,  0.30312938,
         0.24800433, -0.04739454, -0.05923274]]), array([0.0272715])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.9479909 ,  0.98453681,  0.76549318,  0.33478973,  0.25779188,
         0.21220803, -0.05441955,  0.0101878 ]]), array([-0.02412326])]
[array([[-1.9479909 ,  0.98453681,  0.76549318,  0.33478973,  0.25779188,
         0.21220803, -0.05441955,  0.0101878 ]]), array([-0.02412326])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.94835857,  0.98473197,  0.76558812,  0.3348797 ,  0.25788868,
         0.21217689, -0.05453704,  0.0100984 ]]), array([-0.0239643])]
[array([[-1.94835857,  0.98473197,  0.76558812,  0.3348797 ,  0.25788868,
         0.21217689, -0.05453704,  0.0100984 ]]), array([-0.0239643])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.87234171,  0.95117931,  0.74086851,  0.34350098,  0.23251607,
         0.20920638, -0.05163238,  0.01306987]]), array([-0.06679945])]
[array([[-1.87234171,  0.95117931,  0.74086851,  0.34350098,  0.23251607,
         0.20920638, -0.05163238,  0.01306987]]), array([-0.06679945])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.95957777,  1.01230289,  0.77106889,  0.38340782,  0.21349471,
         0.13477718,  0.00885162,  0.02193889]]), array([-0.02596201])]
[array([[-1.95957777,  1.01230289,  0.77106889,  0.38340782,  0.21349471,
         0.13477718,  0.00885162,  0.02193889]]), array([-0.02596201])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96151562,  1.01262071,  0.74709683,  0.39751788,  0.20214313,
         0.11535975,  0.01965082,  0.03113518]]), array([-0.02301428])]
[array([[-1.96151562,  1.01262071,  0.74709683,  0.39751788,  0.20214313,
         0.11535975,  0.01965082,  0.03113518]]), array([-0.02301428])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96122076,  1.01292858,  0.74657095,  0.39780945,  0.20231939,
         0.11495493,  0.01916735,  0.0315517 ]]), array([-0.02342312])]
[array([[-1.96122076,  1.01292858,  0.74657095,  0.39780945,  0.20231939,
         0.11495493,  0.01916735,  0.0315517 ]]), array([-0.02342312])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.9612979 ,  1.01293311,  0.7466081 ,  0.39781828,  0.20233528,
         0.11494453,  0.01916277,  0.0315339 ]]), array([-0.02336991])]
[array([[-1.9612979 ,  1.01293311,  0.7466081 ,  0.39781828,  0.20233528,
         0.11494453,  0.01916277,  0.0315339 ]]), array([-0.02336991])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96129786,  1.01293094,  0.74660913,  0.39781855,  0.20233573,
         0.1149425 ,  0.01916386,  0.03153391]]), array([-0.0233708])]
[array([[-1.96129786,  1.01293094,  0.74660913,  0.39781855,  0.20233573,
         0.1149425 ,  0.01916386,  0.03153391]]), array([-0.0233708])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.9612981 ,  1.01293053,  0.74660943,  0.39781862,  0.20233579,
         0.11494214,  0.01916411,  0.03153382]]), array([-0.02337063])]
[array([[-1.9612981 ,  1.01293053,  0.74660943,  0.39781862,  0.20233579,
         0.11494214,  0.01916411,  0.03153382]]), array([-0.02337063])]
>>>>>>>>>>>>>run time:
0.578125s
